xdsl/dialects/builtin.py

```diff
@@ -62,19 +132,20 @@ def from_string_attr(data: StringAttr) -> SymbolNameAttr:
 
 
 @irdl_attr_definition
-class FlatSymbolRefAttr(ParametrizedAttribute):
-    name = "flat_symbol_ref"
-    data: ParameterDef[StringAttr]
+class SymbolRefAttr(ParametrizedAttribute):
+    name = "symbol_ref"
+    root_reference: ParameterDef[StringAttr]
+    nested_references: ParameterDef[ArrayAttr[StringAttr]]
 
     @staticmethod
     @builder
-    def from_str(data: str) -> FlatSymbolRefAttr:
-        return FlatSymbolRefAttr([StringAttr(data)])
+    def from_str(data: str) -> SymbolRefAttr:
```

Comment:
Don't we want ways to actually build nested symbol refs here? So maybe change the arguments to something like `root: str, nested: List[str] = None`.

Dito for the other constructor.

Reply:
You're right. I'll add this.